### PR TITLE
Add a new Lenovo platform NE2572

### PR DESCRIPTION
**Pull request by psarker**
A new Lenovo x86 platform NE2572

### CODE REVIEW

File: machine/lenovo/lenovo_ne2572/i2ctools/001-i2ctools-sys-eeprom-16bits-address.patch [hunk 1]
```diff
@@ -0,0 +1,24 @@
+diff --git a/sys_eeprom/onie_tlvinfo.c b/sys_eeprom/onie_tlvinfo.c
```
Comment (cbrune) [inline machine/lenovo/lenovo_ne2572/i2ctools/001-i2ctools-sys-eeprom-16bits-address.patch]: For a couple of years now, the ONIE project does not accept the "i2ctools" implementation of onie-syseeprom for new platforms.  We have standardized on the busybox implementation.  Discussed here:

http://lists.opencompute.org/pipermail/opencompute-onie/2015-March/000512.html

Looking around I see I mistakenly accepted this for your `lenovo_ne10032` platform in July of this year.  That was a mistake on my part.  Missed it.

The busybox implementation is much better, more configurable without the need for platform specific patches to change things like the delay timing.  See this example:
https://github.com/opencomputeproject/onie/blob/master/machine/accton/accton_as5812_54x/busybox/conf/config

Also see these commits that added the busybox implementation:
fc76004dca594a5f4f12a7905474f77c67e5effb
481b975ac4c5e8dbec255a2ffb1af3e383475d3a

File: machine/lenovo/lenovo_ne2572/machine.make [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+# Set Linux kernel version
+LINUX_VERSION		= 4.1
+LINUX_MINOR_VERSION	= 38
+
+# Specify uClibc version
+#UCLIBC_VERSION = 0.9.32.1
+GCC_VERSION = 4.9.2
+
+#Extra kernel command line
+EXTRA_CMDLINE_LINUX ?= quiet
```
Comment (cbrune) [inline machine/lenovo/lenovo_ne2572/machine.make]: quiet is already added to the kernel command line

File: machine/lenovo/lenovo_ne2572/machine.make [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+VENDOR_VERSION = .0.3
+
+# Vendor ID -- IANA Private Enterprise Number:
+# http://www.iana.org/assignments/enterprise-numbers
+# Lenovo 
+VENDOR_ID = 19046
+
+# Enable the i2ctools and the onie-syseeprom command for this platform
+I2CTOOLS_ENABLE = yes
+I2CTOOLS_SYSEEPROM = no
```
Comment (cbrune) [inline machine/lenovo/lenovo_ne2572/machine.make]: `I2CTOOLS_SYSEEPROM = no` is the default already.

Funny, with this set to `no` the i2ctools patches above are not used at all.

Also you have defined the busybox implementation of onie-syseeprom.

I think the thing to do here is completely remove the i2ctools directory from this patch.

Comment (cbrune) [reply machine/lenovo/lenovo_ne2572/machine.make]: Same comments for the previous `lenovo_ne10032` platform.  Looks like you can get rid of the `i2ctools` subdirectory.  It is not being used.

File: machine/lenovo/lenovo_ne2572/i2ctools/001-i2ctools-sys-eeprom-16bits-address.patch (hunk 1 above)
Comment (psarker) [reply machine/lenovo/lenovo_ne2572/i2ctools/001-i2ctools-sys-eeprom-16bits-address.patch]: Yea.. it was my mistake.. let me completely remove i2ctool directory

File: machine/lenovo/lenovo_ne2572/machine.make (hunk 2 above)
Comment (psarker) [reply machine/lenovo/lenovo_ne2572/machine.make]: I just have checked it and im still seeing when demo OS boots up it prints kernel msgs in console. So, I will keep this flag ON.

File: machine/lenovo/lenovo_ne2572/machine.make (hunk 3 above)
Comment (psarker) [reply machine/lenovo/lenovo_ne2572/machine.make]: I will comment out I2CTOOLS_SYSEEPROM = no. thank you. I will also delete i2ctools directory

**Comment by psarker** [comment]
Hi Curt,
I have created a new submit #663 to address your comments. I just kept "EXTRA_CMDLINE_LINUX ?= quiet" as it is in machine.make.

Thanks,
Pinaki

File: machine/lenovo/lenovo_ne2572/machine.make (hunk 2 above)
Comment (cbrune) [reply machine/lenovo/lenovo_ne2572/machine.make]: The demo OS does not use `quiet`, you are right.  That was on purpose to see that things were working.

When ONIE boots, it uses `quiet`.

If you want demo OS to also be quiet, then this is fine.

**Comment by cbrune** [comment]
this is handled by #663.

**Pull request closed without merging**